src/bp/lang-server/service/index.ts

```diff
@@ -41,10 +62,34 @@ export default class LanguageService {
     this._models = this._getModels()
     const languages = Object.keys(this._models)
 
+    if (languages.length > 0) {
+      const ramInfos = checkEnoughRam(languages.length, this.dim)
+
+      if (ramInfos.prediction + ramInfos.prediction * MARGIN_PRED > ramInfos.free) {
+        this.logger.warn(
+          'Warning, this program can crash silently if it lacks free memory space. Make you sure you have enough RAM available.'
+        )
```

Comment:
this is duplicated code (here and in the interval. please extract this as a private method

Reply:
Added private `warnRam` method